### PR TITLE
Feat(LSP):  Load pkg files in LSP

### PR DESCRIPTION
**Pull request by He1pa**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [x] N
- [ ] Y 

#### 2. What is the scope of this PR (e.g. component or file name):
LSP

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):
If the input file of lsp is not in a stack environment, load all kcl files in the dir where the file is located as input

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: kclvm/driver/src/lib.rs [hunk 1]
```diff
@@ -111,7 +114,38 @@ pub fn lookup_compile_unit(file: &str) -> (Vec<String>, Option<LoadProgramOption
...
+                                    }
+                                }
+                                if !files.is_empty() {
+                                    return (files, None);
+                                }
+                            }
+                        }
+                    }
+                }
+                return (vec![file.to_string()], None);
```
Comment (Peefy) [inline kclvm/driver/src/lib.rs:143]: The if and else branches seem to have returned the same content.

File: kclvm/driver/src/lib.rs [hunk 2]
```diff
@@ -111,7 +114,38 @@ pub fn lookup_compile_unit(file: &str) -> (Vec<String>, Option<LoadProgramOption
                 Err(_) => return (vec![file.to_string()], None),
             }
         }
-        Err(_) => return (vec![file.to_string()], None),
+        Err(_) => {
+            if load_pkg {
+                let path = Path::new(file);
+                if let Some(ext) = path.extension() {
+                    if ext == KCL_FILE_EXTENSION && path.is_file() {
+                        if let Some(parent) = path.parent() {
```
Comment (Peefy) [inline kclvm/driver/src/lib.rs:122]: use `kclvm_tools::util::get_kcl_files`?

File: kclvm/driver/src/lib.rs (hunk 1 above)
Comment (He1pa) [reply kclvm/driver/src/lib.rs:143]: It will return early if found some pkg files

File: kclvm/driver/src/lib.rs [hunk 3]
```diff
@@ -111,7 +116,23 @@ pub fn lookup_compile_unit(file: &str) -> (Vec<String>, Option<LoadProgramOption
...
+                            if let Ok(files) = get_kcl_files(parent, false){
```
Comment (Peefy) [inline kclvm/driver/src/lib.rs:125]: Format code.

File: kclvm/driver/src/lib.rs [hunk 4]
```diff
@@ -111,7 +116,23 @@ pub fn lookup_compile_unit(file: &str) -> (Vec<String>, Option<LoadProgramOption
...
+                                return (files, None)
+                            }
+                        }
+                    }
+                }
+                return (vec![file.to_string()], None);
+            } else {
```
Comment (Peefy) [inline kclvm/driver/src/lib.rs:132]: Remove the else branch and just use `(vec![file.to_string()], None)`?

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**